### PR TITLE
Improve reliability of WebKit communication bridge

### PR DESCRIPTION
**Pull request by ethan-kusters**
Bug/issue #, if applicable: rdar://103089222

## Summary

Before this change we would send WebKit JSON messages directly by using backtick string literal syntax. This proved problematic since we now need to escape any backticks in the JSON. Instead we can use `JSON.stringify` which _does_ accept a string literal and convert that literal into a JSON object before calling `JSON.parse`

This removes the need for any escaping and fixes some edge cases in WebKit communication.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by ethan-kusters** [comment]
@swift-ci please test

File: Sources/SwiftDocC/Infrastructure/Communication/WebKitCommunicationBridge.swift [hunk 1]
```diff
@@ -49,10 +49,8 @@ public struct WebKitCommunicationBridge: CommunicationBridge {
         do {
             let encodedMessage = try JSONEncoder().encode(message)
             let messageJSON = String(data: encodedMessage, encoding: .utf8)!
-                // Escape backticks to avoid conflicting with JavaScript's template string syntax.
-                .replacingOccurrences(of: "`", with: "\\`")
             
-            evaluateJavaScript("window.bridge.receive(JSON.parse(`\(messageJSON)`))") { _, _ in }
+            evaluateJavaScript("window.bridge.receive(JSON.parse(JSON.stringify(\(messageJSON))))") { _, _ in }
```
Comment (franklinsch) [inline Sources/SwiftDocC/Infrastructure/Communication/WebKitCommunicationBridge.swift:53]: Doesn't this mean that what we're passing to `window.bridge.receive` is a string instead of an object now? Since DocC-Render expects an object, I suspect this change will break things?

Comment (dobromir-hristov) [reply Sources/SwiftDocC/Infrastructure/Communication/WebKitCommunicationBridge.swift:53]: No. `JSON.prase` parses it to an object.

Comment (franklinsch) [reply Sources/SwiftDocC/Infrastructure/Communication/WebKitCommunicationBridge.swift:53]: Aah I see, the confusion for me was that `messageJSON` isn't actually a JavaScript string literal, it's a JSON object. In that case why not pass it directly with `window.bridge.receive(\(messageJSON))`?

Comment (dobromir-hristov) [reply Sources/SwiftDocC/Infrastructure/Communication/WebKitCommunicationBridge.swift:53]: was just about to ask :)

**Review by dobromir-hristov — COMMENTED** [review]
I think we can send the JSON directly over the bridge without parsing back and forth.

File: Sources/SwiftDocC/Infrastructure/Communication/WebKitCommunicationBridge.swift (hunk 1 above)
Comment (ethan-kusters) [reply Sources/SwiftDocC/Infrastructure/Communication/WebKitCommunicationBridge.swift:53]: Ah right good point... Curious why that wasn't done originally then? But yes that seems to work as well – I'll update the PR.

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Review by hqhhuang — APPROVED** [review]
Didn't know that `messageJSON` was already formatted correctly 😁

**Review by dobromir-hristov — APPROVED** [review]

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Comment by ethan-kusters** [comment]
@swift-ci please test macOS platform
